srtp/srtp.c

```diff
@@ -2074,6 +2060,154 @@ srtp_remove_stream(srtp_t session, uint32_t ssrc) {
 }
 
 
+srtp_err_status_t
+srtp_update(srtp_t session, const srtp_policy_t *policy) {
+  srtp_err_status_t stat;
+
+  /* sanity check arguments */
+  if ((session == NULL) || (policy == NULL) || (policy->key == NULL))
+    return srtp_err_status_bad_param;
+
+  while (policy != NULL) {
+    stat = srtp_update_stream(session, policy);
+    if (stat) {
+      return stat;
+    }
+
+    /* set policy to next item in list  */
+    policy = policy->next;
+  }
+  return srtp_err_status_ok;
+}
+
+
+srtp_err_status_t
+srtp_update_stream(srtp_t session, const srtp_policy_t *policy) {
+  srtp_err_status_t status;
+
+  /* sanity check arguments */
+  if ((session == NULL) || (policy == NULL) || (policy->key == NULL))
+    return srtp_err_status_bad_param;
+
+  switch (policy->ssrc.type) {
+  case (ssrc_any_outbound):
+  case (ssrc_any_inbound): {
+    if (session->stream_template == NULL) {
+      return srtp_err_status_bad_param;
+    }
+
+    srtp_stream_t new_stream_template;
+    srtp_stream_t new_stream_list = NULL;
+
+    /* allocate new template stream  */
+    status = srtp_stream_alloc(&new_stream_template, policy);
+    if (status) {
+      return status;
+    }
+
+    /* initialize new template stream  */
+    status = srtp_stream_init(new_stream_template, policy);
+    if (status) {
+      srtp_crypto_free(new_stream_template);
+      return status;
+    }
+
+    /* for all old templated streams */
+    for (;;) {
+      srtp_stream_t stream = session->stream_list;
+      while ((stream != NULL) && (stream->rtp_auth != session->stream_template->rtp_auth)) {
+        stream = stream->next;
+      }
+      if (stream == NULL) {
+        /* no more templated streams */
+        break;
+      }
+
+      uint32_t ssrc = stream->ssrc;
+      /* save old extendard seq */
+      srtp_xtd_seq_num_t old_index = stream->rtp_rdbx.index;
+
+      /* remove stream */
+      status = srtp_remove_stream(session, ssrc);
+      if (status) {
+        /* free new allocations */
+        while (new_stream_list != NULL) {
+          srtp_stream_t next = new_stream_list->next;
+          status = srtp_stream_dealloc(new_stream_list, new_stream_template);
+          if (status)
```

Comment:
You should include braces around the conditional block of code for any if statement.  The exception to this rule is when it can all be contained on a single line.  This is a coding standard that has been in place at Cisco for a long time.